Test/Microsoft.ApplicationInsights.Test/Shared/Extensibility/TelemetryConfigurationTest.cs

```diff
@@ -285,6 +288,210 @@ public void InstrumentationKeyCanBeSetToProgrammaticallyDefineInstrumentationKey
+            {
+                ConnectionString = connectionString
+            };
+            
+            Assert.AreEqual(explicitEndpoint, configuration.Endpoint.Ingestion.AbsoluteUri);
+        }
+
+        [TestMethod]
+        [TestCategory("ConnectionString")]
+        public void VerifySetConnectionString_SetsChannelDefaultEndpoint()
+        {
+            var connectionString = $"InstrumentationKey=00000000-0000-0000-0000-000000000000";
+
+            var channel = new InMemoryChannel();
+
+            var configuration = new TelemetryConfiguration
+            {
+                TelemetryChannel = channel,
+                ConnectionString = connectionString,
+            };
+
+            Assert.AreEqual("https://dc.services.visualstudio.com/", configuration.Endpoint.Ingestion.AbsoluteUri);
+            Assert.AreEqual("https://dc.services.visualstudio.com/v2/track", channel.EndpointAddress);
+        }
+
+        [TestMethod]
+        [TestCategory("ConnectionString")]
+        public void VerifySetConnectionString_SetsChannelExpliticEndpoint()
+        {
+            var explicitEndpoint = "https://127.0.0.1/";
+            var connectionString = $"InstrumentationKey=00000000-0000-0000-0000-000000000000;IngestionEndpoint={explicitEndpoint}";
+
+            var channel = new InMemoryChannel();
+
+            var configuration = new TelemetryConfiguration
+            {
+                TelemetryChannel = channel,
+                ConnectionString = connectionString,
+            };
+
+            Assert.AreEqual(explicitEndpoint, configuration.Endpoint.Ingestion.AbsoluteUri);
+            Assert.AreEqual(explicitEndpoint + "v2/track", channel.EndpointAddress);
+        }
+
+        [TestMethod]
+        [TestCategory("ConnectionString")]
+        public void E2E_DefaultScenario()
+        {
+            var tConfig = new TelemetryConfiguration();
+
+            Assert.AreEqual("https://dc.services.visualstudio.com/v2/track", tConfig.DefaultTelemetrySink.TelemetryChannel.EndpointAddress);
+        }
+
+        [TestMethod]
+        [TestCategory("ConnectionString")]
+        public void E2E_DefaultScenario_ConfigurationConstructor()
+        {
+            var tConfig = new TelemetryConfiguration("00000000-0000-0000-0000-000000000000", new InMemoryChannel());
+
+            Assert.AreEqual("https://dc.services.visualstudio.com/v2/track", tConfig.DefaultTelemetrySink.TelemetryChannel.EndpointAddress);
+        }
+
+        [TestMethod]
+        [TestCategory("ConnectionString")]
+        public void E2E_DefaultScenario_WithConnectionString()
+        {
+            var tConfig = new TelemetryConfiguration
+            {
+                ConnectionString = "InstrumentationKey=00000000-0000-0000-0000-000000000000;IngestionEndpoint=https://127.0.0.1/"
+            };
+
+            Assert.AreEqual("https://127.0.0.1/v2/track", tConfig.DefaultTelemetrySink.TelemetryChannel.EndpointAddress);
+        }
```

Comment:
@lmolkova , @cijothomas , @Dmitry-Matveev 
I've got 3 "E2E" examples here of creating a TelemetryConfiguration and setting the TelemteryChannel.
Can you think of any scenarios that I might have missed?